### PR TITLE
Following Filter

### PR DESCRIPTION
**Pull request by Specture7**
RePR from the last one

### CODE REVIEW

File: src/jvmMain/kotlin/io/beatmaps/api/search.kt [hunk 1]
```diff
@@ -197,6 +200,16 @@ fun Route.searchRoute() {
         val sortArgs = searchInfo.sortArgsFor(actualSortOrder)
 
         newSuspendedTransaction {
+            val followingSubQuery = if (it.followed == true) {
+                requireAuthorization { _, user ->
```
Comment (Top-Cat) [inline src/jvmMain/kotlin/io/beatmaps/api/search.kt:204]: This shouldn't be a call to requireAuthorization, it very much seems that when not logged in you just ignore the property.
The call has the side effect of returning an error to the user when not logged in.

Just wrap the whole handler in optionalAuthorization and check if the user is null / auth type is None

Comment (Top-Cat) [reply src/jvmMain/kotlin/io/beatmaps/api/search.kt:204]: (Which will mean you won't need those changes to `requireAuthorization` and `optionalAuthorization`)

**Review by Top-Cat — CHANGES_REQUESTED** [review]

**Comment by Top-Cat** [comment]
But ok to test

**Comment by Specture7** [comment]
Should be good now

File: src/jvmMain/kotlin/io/beatmaps/api/search.kt [hunk 2]
```diff
@@ -187,108 +191,129 @@ fun Route.searchRoute() {
...
-                                        .notNull(it.curator) { o -> Beatmap.curator eq o }
-                                }
-                                .orderBy(*sortArgs)
-                                .limit(it.page)
+                if (searchInfo.escapedQuery != null && searchInfo.escapedQuery.startsWith("key:")) {
+                    Beatmap
+                        .slice(Beatmap.id)
+                        .select {
+                            Beatmap.id eq searchInfo.escapedQuery.substring(4)
+                                .toInt(16) and (Beatmap.deletedAt.isNull())
```
Comment (Top-Cat) [inline src/jvmMain/kotlin/io/beatmaps/api/search.kt:218]: That's a whacky place to add a linebreak (does it even need one?)

File: src/jvmMain/kotlin/io/beatmaps/api/search.kt [hunk 3]
```diff
@@ -187,108 +191,129 @@ fun Route.searchRoute() {
...
+                        .select {
+                            Beatmap.id.inSubQuery(
+                                Beatmap
+                                    .joinUploader()
+                                    .crossJoin(
+                                        Versions
+                                            .let { q ->
+                                                if (needsDiff) {
+                                                    q.join(
+                                                        Difficulty,
```
Comment (Top-Cat) [inline src/jvmMain/kotlin/io/beatmaps/api/search.kt:247]: How small is your screen? 🙈

**Comment by Specture7** [comment]
fixed the style a bit, i hope this one is good

**Pull request merged**